libs/shared/ui/src/lib/components/link/link.tsx

```diff
@@ -1,58 +1,78 @@
-import { NavLink } from 'react-router-dom'
-import { type IconEnum } from '@qovery/shared/enums'
+import { type VariantProps, cva } from 'class-variance-authority'
+import { type ComponentPropsWithoutRef, forwardRef } from 'react'
+import { Link as ReactLink, type LinkProps as ReactLinkProps } from 'react-router-dom'
+import { twMerge } from 'tailwind-merge'
 import Icon from '../icon/icon'
+import { IconAwesomeEnum } from '../icon/icon-awesome.enum'
 
-export interface BaseLink {
-  link: string
-  linkLabel?: string
-  external?: boolean
-}
+const linkVariants = cva(
+  ['cursor-pointer', 'transition', 'duration-100', 'font-medium', 'inline-flex', 'flex-center', 'gap-1'],
+  {
+    variants: {
+      color: {
+        brand: ['text-brand-500', 'hover:text-brand-600'],
+        red: ['text-red-500', 'hover:text-red-600'],
+        sky: ['text-sky-500', 'hover:text-sky-600'],
+      },
+      size: {
+        xs: ['text-xs'],
+        sm: ['text-sm'],
+      },
+    },
+    defaultVariants: {
+      color: 'sky',
+      size: 'sm',
+    },
+  }
+)
 
-export interface LinkProps extends BaseLink {
+const iconVariants = cva([], {
+  variants: {
+    size: {
+      xs: ['text-2xs', 'leading-4'],
+      sm: ['text-xs', 'leading-5'],
+    },
+  },
+  defaultVariants: {
+    size: 'sm',
+  },
+})
+
+export interface ExternalLinkProps
+  extends Omit<ComponentPropsWithoutRef<'a'>, 'color'>,
+    VariantProps<typeof linkVariants> {
```

Comment:
You don't have to declare `className`

Reply:
Fixed https://github.com/Qovery/console/pull/864/commits/a4c635fccd79ce0ea29060a8cb2c673d221b9418